### PR TITLE
With usrsctp

### PR DESCRIPTION
**Pull request by ruengeler**
Support for usrsctp.

### CODE REVIEW

File: neat_he.c [hunk 1]
```diff
@@ -176,11 +165,14 @@ he_resolve_cb(struct neat_resolver *resolver, struct neat_resolver_results *resu
...
         callback_fx = resolver->userData2;
-
         if (flow->connectfx(he_ctx, callback_fx) == -1) {
+            neat_log(NEAT_LOG_DEBUG, "%s: Connect failed", __func__);
             continue;
         } else {
+            neat_log(NEAT_LOG_DEBUG, "%s: Connect successful", __func__);
             flow->heConnectAttemptCount++;
+            if (candidate->ai_protocol == IPPROTO_SCTP)
+                break;
```
Comment (tuexen) [inline neat_he.c]: What makes this SCTP special thing necessary?

File: neat_core.c [hunk 2]
```diff
@@ -561,15 +642,14 @@ he_connected_cb(uv_poll_t *handle, int status, int events)
...
+//#endif
 
         // TODO: Security layer.
-
         uvpollable_cb(handle, NEAT_OK, UV_WRITABLE);
     } else {
 
         /* TODO: Used by Karl-Johan Grinnemo during test. Remove in final version. */
-#if 0
+//#if 0
```
Comment (tuexen) [inline neat_core.c]: Why this change?

File: neat_core.c [hunk 3]
```diff
@@ -579,11 +659,11 @@ he_connected_cb(uv_poll_t *handle, int status, int events)
                 ip_address,
                 INET_ADDRSTRLEN, 0, 0, NI_NUMERICHOST);
         printf("Loosing connection attempt to %s with protocol %d\n", ip_address, flow->sockProtocol);
-#endif
+//#endif
```
Comment (tuexen) [inline neat_core.c]: Why this change?

File: neat_core.c [hunk 4]
```diff
@@ -551,7 +632,7 @@ he_connected_cb(uv_poll_t *handle, int status, int events)
         free(he_ctx);
 
         /* TODO: Used by Karl-Johan Grinnemo during test. Remove in final version. */
-#if 0
+//#if 0
```
Comment (tuexen) [inline neat_core.c]: Why this change?

File: neat_core.c [hunk 5]
```diff
@@ -561,15 +642,14 @@ he_connected_cb(uv_poll_t *handle, int status, int events)
                     ip_address,
                     INET_ADDRSTRLEN, 0, 0, NI_NUMERICHOST);
         printf("Winning connection attempt to %s with protocol %d\n", ip_address, flow->sockProtocol);
-#endif
+//#endif
```
Comment (tuexen) [inline neat_core.c]: Why this change?

**Comment by weinrank** [comment]
FYI: I changed the CMake file to not include usrsctp support by default -  c394c55c3def570ea71e397481b7001f55015054

We have compile errors on netbsd and ubuntu:
http://buildbot.nplab.de:28010/builders/netbsd/builds/14/steps/compile/logs/stdio
http://buildbot.nplab.de:28010/builders/ubuntu-16.04/builds/17/steps/compile/logs/stdio
